src/navigation.ts

```diff
@@ -1294,55 +1294,90 @@ export class Navigation {
   }
 
   /**
-   * Show the action menu for a given node.
+   * Show the action menu for the current node.
    *
    * The action menu will contain entries for relevant actions for the
    * node's location.  If the location is a block, this will include
    * the contents of the block's context menu (if any).
+   *
+   * Returns true if it is possible to open the action menu in the
+   * current location, even if the menu was not opened due there being
+   * no applicable menu items.
    */
-  openActionMenu(node: Blockly.ASTNode) {
-    const fakeEvent = fakeEventForNode(node);
-    (node.getLocation() as Blockly.BlockSvg).showContextMenu(fakeEvent);
-
+  openActionMenu(workspace: Blockly.WorkspaceSvg): boolean {
     let menuOptions: Array<
       | Blockly.ContextMenuRegistry.ContextMenuOption
       | Blockly.ContextMenuRegistry.LegacyContextMenuOption
-    > | null = null;
+    > = [];
     let rtl: boolean;
-    let workspace: Blockly.WorkspaceSvg;
 
+    const cursor = workspace.getCursor();
+    if (!cursor) throw new Error('workspace has no cursor');
+    const node = cursor.getCurNode();
     const nodeType = node.getType();
     switch (nodeType) {
       case Blockly.ASTNode.types.BLOCK:
         const block = node.getLocation() as Blockly.BlockSvg;
-        workspace = block.workspace as Blockly.WorkspaceSvg;
         rtl = block.RTL;
-
         // Reimplement BlockSvg.prototype.generateContextMenu as that
         // method is protected.
-        if (!workspace.options.readOnly && !block.contextMenu) {
+        if (!workspace.options.readOnly && block.contextMenu) {
           menuOptions =
             Blockly.ContextMenuRegistry.registry.getContextMenuOptions(
               Blockly.ContextMenuRegistry.ScopeType.BLOCK,
               {block},
             );
 
           // Allow the block to add or modify menuOptions.
-          if (block.customContextMenu) {
-            block.customContextMenu(menuOptions);
-          }
+          block.customContextMenu?.(menuOptions);
         }
         // End reimplement.
         break;
+
+      // case Blockly.ASTNode.types.INPUT:
+      case Blockly.ASTNode.types.NEXT:
+      case Blockly.ASTNode.types.PREVIOUS:
+        const connection = node.getLocation() as Blockly.Connection;
+        rtl = connection.getSourceBlock().RTL;
+
+        // Slightly hacky: get insert action from registry.  Hacky
```

Comment:
I added a note in https://github.com/google/blockly-keyboard-experimentation/issues/155 to add connection as a scope type.